src/python/dxpy/bindings/dxfile.py

```diff
@@ -586,6 +586,11 @@ def upload_part(self, data, index=None, display_progress=False, report_progress_
         defaults to 1. This probably only makes sense if this is the
         only part to be uploaded.
         """
+    
+        # AZURE-314: Do not upload a first (single) empty part if region settings do not allow that
+        if index == 1 and len(data) == 0 and not self._empty_last_part_allowed:
```

Comment:
Would it suffice to add this logic in self.close? It it makes sense, I think it would be a better place for this since (1) there is already a check there to detect the "special case" in S3 of an empty file being uploaded, which seems to be analogous to the case you are trying to handle here and (2) it preserves the property that every invocation of this method uploads exactly one part.